xds/internal/xdsclient/xdsresource/unmarshal_cds_test.go

```diff
@@ -1620,3 +1622,181 @@ func (s) TestUnmarshalCluster(t *testing.T) {
 		})
 	}
 }
+
+func (s) TestValidateClusterWithOutlierDetection(t *testing.T) {
+	oldOutlierDetectionSupportEnv := envconfig.XDSOutlierDetection
+	envconfig.XDSOutlierDetection = true
+	defer func() { envconfig.XDSOutlierDetection = oldOutlierDetectionSupportEnv }()
+	odProto := func(od *v3clusterpb.OutlierDetection) *v3clusterpb.Cluster {
```

Comment:
This function returns a `Cluster` proto, so something like `odToClusterProto` might be a better name (or something else if you have other ideas).

Reply:
Ah, good point. Switched.